distribution/push_v2.go

```diff
@@ -390,7 +390,7 @@ func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.
 		}
 
 		// when error is unauthorizedError and user don't hasAuthInfo that's the case user don't has right to push layer to register
-		// and he hasn't login either, in this case candidate cache should be removed
+		// and they hasn't login either, in this case candidate cache should be removed
```

Comment:
Yup, this one needs some touching up (but we should avoid using pronouns here at all; not needed for the description)